### PR TITLE
feat(business_profile): add business profile table and CRUD endpoints

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] New feature

## Description

This PR adds the tables required for supporting business profiles 

### Additional Changes

- [x] This PR modifies the API contract
5 endpoints are added. Business profile create, retrieve, update, delete
- [x] This PR modifies the database schema
Migrations can be found [here](https://github.com/juspay/hyperswitch/blob/3115c2fbe1aa720148867ec98b82826a3632e77c/migrations/2023-08-08-144148_add_business_profile_table/up.sql)

## How did you test it?

- First create a merchant account.
- Create a business profile in merchant account.
```bash
curl --location 'http://localhost:8080/account/merchant_1691666029/business_profile' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{}'
```

Screenshot 2023-08-11 at 5 19 29 PM

- Retrieve a business profile
```bash
curl --location 'http://localhost:8080/account/merchant_1691666029/business_profile/pro_bNkCVL60jFUYUhcVqe7g' \
--header 'api-key: test_admin'
```
Screenshot 2023-08-11 at 5 19 47 PM

- Update a business profile
```bash
curl --location 'http://localhost:8080/account/merchant_1691666029/business_profile/pro_bNkCVL60jFUYUhcVqe7g' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "profile_name": "adyen_uk"
}'
```
Screenshot 2023-08-11 at 5 20 03 PM

- List business profiles
```bash
curl --location 'http://localhost:8080/account/merchant_1692340382/business_profile' \
--header 'api-key: test_admin'
```

- Delete a business profile
```bash
curl --location --request DELETE 'http://localhost:8080/account/merchant_1691666029/business_profile/pro_bNkCVL60jFUYUhcVqe7g' \
--header 'api-key: test_admin'
```

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/diesel_models/src/business_profile.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    serde::Serialize,
+    Identifiable,
+    Queryable,
+    router_derive::DebugAsDisplay,
+)]
+#[diesel(table_name = business_profile, primary_key(profile_id))]
+pub struct BusinessProfile {
+    pub profile_id: String,
+    pub merchant_id: String,
+    pub profile_name: Option<String>,
```
Comment (jarnura) [inline crates/diesel_models/src/business_profile.rs:19]: Remove optional value for this.

File: crates/diesel_models/src/business_profile.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    pub payout_routing_algorithm: Option<serde_json::Value>,
+    pub is_recon_enabled: bool,
+}
+
+#[derive(Clone, Debug, Insertable, router_derive::DebugAsDisplay)]
+#[diesel(table_name = business_profile, primary_key(profile_id))]
+pub struct BusinessProfileNew {
+    pub profile_id: String,
+    pub merchant_id: String,
+    pub profile_name: Option<String>,
```
Comment (jarnura) [inline crates/diesel_models/src/business_profile.rs:40]: Remove optional value for this.

File: crates/router/src/core/admin.rs [hunk 3]
```diff
@@ -830,6 +831,218 @@ pub fn get_frm_config_as_secret(
     }
 }
 
+pub async fn create_business_profile(
+    db: &dyn StorageInterface,
+    request: api::BusinessProfileCreate,
+    merchant_id: &str,
+) -> RouterResponse<api_models::admin::BusinessProfileResponse> {
+    let key_store = db
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:839]: Make the db calls parallel.

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/lib.rs [hunk 4]
```diff
@@ -111,8 +111,9 @@ pub fn mk_app(
             .service(routes::Customers::server(state.clone()))
             .service(routes::Configs::server(state.clone()))
             .service(routes::Refunds::server(state.clone()))
+            .service(routes::BusinessProfile::server(state.clone()))
             .service(routes::MerchantConnectorAccount::server(state.clone()))
-            .service(routes::Mandates::server(state.clone()));
+            .service(routes::Mandates::server(state.clone()))
```
[the remark this answers is not in the document]
Comment (Narayanbhat166) [reply crates/router/src/lib.rs:116]: it seems like the semicolon is not necessary, as it compiles

File: crates/router/src/core/admin.rs (hunk 3 above)
Comment (Narayanbhat166) [reply crates/router/src/core/admin.rs:839]: find_merchant_account depends on keystore, cannot make these parallel

**Review by jarnura — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**